File: src/main/java/de/spries/fleetcommander/model/core/universe/TurnEventBus.java
package de.spries.fleetcommander.model.core.universe;

import de.spries.fleetcommander.model.core.Player;

public interface TurnEventBus {

	void fireConqueredEnemyPlanet(Player invader);

	void fireConqueredUninhabitedPlanet(Player invader);

	void fireDefendedPlanet(Player inhabitant);

	void fireLostPlanet(Player previousInhabitant);

	void fireLostShipFormation(Player commander);
}

File: src/main/java/de/spries/fleetcommander/model/core/universe/TurnEventBus.kt
package de.spries.fleetcommander.model.core.universe

import de.spries.fleetcommander.model.core.Player

interface TurnEventBus {

    fun fireConqueredEnemyPlanet(invader: Player)

    fun fireConqueredUninhabitedPlanet(invader: Player)

    fun fireDefendedPlanet(inhabitant: Player)

    fun fireLostPlanet(previousInhabitant: Player)

    fun fireLostShipFormation(commander: Player)
}
